src/fsharp/absil/il.fs

```diff
@@ -1030,6 +1030,16 @@ type ILAttributesStored =
        | Reader f -> ILAttributes (f metadataIndex)
        | Given attrs -> attrs
 
+[<Struct>]
+type ILAttributesStoredWithIndex =
```

Comment:
Any reason not to write this with a primary constructor for the Struct?

Reply:
No, just copied the approach from `ILVersionInfo` without thinking about a better way to do it